dateutil/parser/isoparser.py

```diff
@@ -159,7 +159,7 @@ def parse_isodate(self, datestr):
         components, pos = self._parse_isodate(datestr)
         if pos < len(datestr):
             raise ValueError('String contains unknown ISO ' +
-                             'components: {}'.format(datestr))
+                             'components: {}'.format(datestr.decode('ascii')))
```

Comment:
Nit: I think it'd be even easier to read if we passed here: `'components: {!r}'`.

That will put `'` around the text that failed (Say for example that we have [some invisible characters](https://support.sas.com/resources/papers/proceedings/proceedings/sugi31/055-31.pdf).

Reply:
Fixed: https://github.com/dateutil/dateutil/pull/1122/commits/f28e639ddcbaae341da5472a783381330ca67e0f

Definitely agree with removing the quotes making it more difficult to detect things like whitespace.